### PR TITLE
Temporarily consider 429s as success in links check

### PR DESCRIPTION
**Pull request by dmathieu**
This considers 429s as a success, until we can get lychee to cache success requests even if there's a failure.
This should only be a temporary measure. But ignoring one status code seems better than entirely disabling the check.

Related: #6183.

### CODE REVIEW

File: .github/workflows/links-fail-fast.yml [hunk 1]
```diff
@@ -25,4 +25,4 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           fail: true
-          args: --max-concurrency 5 --cache --max-cache-age 1d .
+          args: --max-concurrency 5 --cache --max-cache-age 1d --accept 100..=103,200..=299,429 .
```
Comment (dashpole) [inline .github/workflows/links-fail-fast.yml:28]: can you add a TODO to remove the accept arg, since this is temporary?

Comment (dashpole) [reply .github/workflows/links-fail-fast.yml:28]: Just want to make sure we don't forget to revert this.

Comment (XSAM) [reply .github/workflows/links-fail-fast.yml:28]: And, add this issue link to the TODO comment: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6183

**Review by dashpole — APPROVED** [review]

**Review by XSAM — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by dmathieu** [comment]
The remaining failure is due to the recent lychee action upgrade, not the 429s.
Merging this, and I'll look into fixing the v2 issue in another PR.

**Pull request merged**